CMakeLists.txt

```diff
@@ -136,6 +137,25 @@ if(BUILD_WITH_VECTORIZATION_SUPPORT)
   list(APPEND EXPORTED_TARGETS_LIST proxsuite-vectorized)
 endif()
 
+if(SERIALIZATION_QP_MODEL)
+  set(cereal_dir ${PROJECT_SOURCE_DIR}/external/cereal)
+  set(cereal ${cereal_dir}/README.md)
+  if(NOT EXISTS ${cereal})
+    message("Initialize cereal submodule.")
+    execute_process(
+      COMMAND ${GIT_EXECUTABLE} submodule update --init ${cereal_dir}
+      WORKING_DIRECTORY ${cereal_dir} COMMAND_ERROR_IS_FATAL ANY)
+  endif()
+
+  target_include_directories(
+    proxsuite
+    INTERFACE
+      "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/external/cereal/include>"
+      "$<INSTALL_INTERFACE:external/cereal/include>")
+  target_compile_definitions(proxsuite INTERFACE PROXSUITE_WITH_SERIALIZATION)
+
+endif()
```

Comment:
We should not install Cereal. We can check if it is present and download for Python bindings and tests.
We should not link the main exported target to cereal. 
Only if the user needs it, it should install it and linked to it.

Reply:
I do not really understand your comment. 
>Only if the user needs it, it should install it and linked to it.

This is what I am doing here I think. I set a CMake Option, that people can turn on if they want. If they turn this on, I make sure Cereal is present, I have to link the include directories etc...